products/bridge/smart-contracts/contracts/Bridged.sol

```diff
@@ -9,37 +9,83 @@ import "./Relayer.sol";
 abstract contract Bridged is Initializable {
     Relayer private _relayer;
 
-    function initialize(Relayer relayer) public initializer {
-        _relayer = relayer;
-    }
+    error NotRelayer(address relayer);
 
     modifier onlyRelayer() {
-        require(msg.sender == address(_relayer), "Must be called by relayer");
+        if (msg.sender != address(_relayer)) {
+            revert NotRelayer(msg.sender);
+        }
         _;
     }
 
-    function dispatched(
+    function __Bridged_init(Relayer relayer_) public onlyInitializing {
+        _relayer = relayer_;
+    }
+
+    function relayer() public view returns (Relayer) {
+        return _relayer;
+    }
+
+    function relay(
+        uint targetChainId,
+        address target,
+        bytes memory call,
+        bool readonly,
+        bytes4 callback
+    ) internal returns (uint nonce) {
+        nonce = _relayer.relay(targetChainId, target, call, readonly, callback);
+    }
+
+    function _dispatched(
         address target,
         bytes memory call
-    ) public payable onlyRelayer returns (bool success, bytes memory response) {
-        console.log("dispatched()");
+    ) internal onlyRelayer returns (bool success, bytes memory response) {
         (success, response) = target.call{value: msg.value, gas: 100000}(call);
     }
 
+    function dispatched(
+        uint targetChainId,
+        address target,
+        bytes memory call
+    )
+        public
+        payable
+        virtual
+        onlyRelayer
+        returns (bool success, bytes memory response)
+    {
+        (success, response) = _dispatched(target, call);
```

Comment:
I had named this argument incorrectly. It should be `sourceChainId` being passed here. The `relayer` passes the `sourceChainId` to the `dispatched` in case it needs to check where the message comes from. However, the target chain id is already verified through the relayer when validating the signature: [bytes memory message = abi.encode(](https://github.com/Zilliqa/zilliqa-developer/blob/998527e831797225f422a6d3b94365f7673158d1/products/bridge/smart-contracts/contracts/Relayer.sol#L128-L137)`